### PR TITLE
Added pre-main e2e jobs with lifecycle-manager

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Replaced github action for `Wait for the 'pull-nats-manager-build' job to succeed` because the old one was not working.
- Added pre-main e2e job with lifecycle-manager

**Related issue(s)**

- https://github.com/kyma-project/nats-manager/issues/76

### CODE REVIEW

File: hack/get_module_template_from_build_job.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+## This script requires the following env variables:
+# PR_NUMBER (e.g. 82)
+# COMMIT_STATUS_JSON
+# PROJECT_ROOT (e.g. "../")
```
Comment (marcobebway) [inline hack/get_module_template_from_build_job.sh:6]: If one of these env vars is missing will there be a meaningful error message?

Comment (mfaizanse) [reply hack/get_module_template_from_build_job.sh:6]: Not yet, but I will validate them in next PR where I need to modify this script for post-main job.

**Review by marcobebway — APPROVED** [review]

**Pull request merged**